misc-utils/findmnt.c

```diff
@@ -1216,8 +1265,12 @@ static int add_matching_lines(struct libmnt_table *tb,
 	while((fs = get_next_fs(tb, itr, findmnt))) {
 		if ((findmnt->flags & FL_TREE) || (findmnt->flags & FL_SUBMOUNTS))
 			rc = create_treenode(table, tb, fs, NULL, findmnt);
-		else
-			rc = !add_line(table, fs, NULL, findmnt);
+		else {
+			struct libscols_line *l = add_line(table, fs, NULL, findmnt);
+			if (l == filtered_line)
+				continue;
```

Comment:
Would it be better to add a "filtered" boolean argument to add_line()
```
   bool filtered = 0;
   struct libscols_line *l = add_line(table, fs, NULL, findmnt, &filtered);
   if (filtered)
     continue;
```

... rather than use a magic global variable  `filtered_line` ?

Reply:
(Fixed locally).